[compression-dictionary] Moved the dictionary hash into the body (#2784)
httpwg/http-extensions: draft-ietf-httpbis-compression-dictionary.md

# Dictionary-Compressed Brotli

The "dcb" content encoding identifies a resource that is a
"Dictionary-Compressed Brotli" stream.

> A "Dictionary-Compressed Brotli" stream is a binary stream that starts with a
> 36-byte header and is followed by a Brotli {{RFC7932}} stream of the response
> that has been compressed with an external dictionary using a compression window
> not larger than 16 MB.

Reviewer: I think that it's about this point that you might acknowledge that describing a thing completely in one sentence is hurting readability a bit.

```suggestion
A "Dictionary-Compressed Brotli" stream has a fixed header that is followed by
a Brotli {{RFC7932}} stream. The header consists of a fixed 4 byte sequence
and a 32 byte hash of the external dictionary that was used.  The Brotli stream
is created using the referenced external dictionary and a compression window
that is at most 16 MB in size.
```

The 36-byte stream header consists of a 4-byte signature followed by the
32-byte SHA-256 hash of the external dictionary that was used to compress the
resource:

|  Field             | Size       |
|--------------------|------------|
|  Magic_Number      | 4 bytes    |

> |  SHA_256_Hash      | 32 bytes   |

Reviewer: I'm not sure if the table really gets you much.

You might instead consider using a definition list, as I suggest below.

> Magic_Number: 4 bytes, little-endian format.  Value: 0x424344FF

Reviewer: ```suggestion
Magic_Number: 4 fixed bytes: 0xff, 0x44, 0x43, 0x42.
```

> Magic_Number: 4 bytes, little-endian format.  Value: 0x424344FF
> SHA_256_Hash: 32 Bytes. SHA-256 hash digest of the dictionary {{SHA-256}}.

Reviewer: ```suggestion
Magic_Number: 
: 4 fixed bytes: 0xff, 0x44, 0x43, 0x42.

SHA_256_Hash:
: 32 Bytes. SHA-256 hash digest of the dictionary {{SHA-256}}.
```

# Dictionary-Compressed Zstandard

The "dcz" content encoding identifies a resource that is a
"Dictionary-Compressed Zstandard" stream.

A "Dictionary-Compressed Zstandard" stream is a binary stream that starts with a
36-byte header and is followed by a Zstandard {{RFC8878}} stream
of the response that has been compressed with an external dictionary.

The 36-byte stream header consists of a 4-byte signature followed by the
32-byte SHA-256 hash of the external dictionary that was used to compress the
resource:

|  Field             | Size       |
|--------------------|------------|
|  Magic_Number      | 4 bytes    |
|  SHA_256_Hash      | 32 bytes   |

> Magic_Number: 4 bytes, little-endian format.  Value: 0x5A4344FF

Reviewer: ```suggestion
Magic_Number: 4 fixed bytes: 0xff, 0x44, 0x45, 0x5a.
```

"Dictionary-Compressed Zstandard" stream.

A "Dictionary-Compressed Zstandard" stream is a binary stream that starts with a
40-byte fixed header and is followed by a Zstandard {{RFC8878}} stream
of the response that has been compressed with an external dictionary.

The 40-byte header consists of a fixed 8-byte sequence followed by the
32-byte SHA-256 hash of the external dictionary that was used to compress the
resource:

Magic_Number:

> : 8 fixed bytes: 0x5e, 0x2a, 0x4d, 0x18, 0x20, 0x00, 0x00, 0x00.

Reviewer 2: LGTM!
